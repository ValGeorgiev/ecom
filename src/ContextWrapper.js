import React, { createContext } from 'react'

export const AuthContext = createContext({ auth: false })

const ContextWrapper = props => {
  return (
    <AuthContext.Provider value={{auth: true}}>
      {props.children}
    </AuthContext.Provider>
  )
}

export default ContextWrapper