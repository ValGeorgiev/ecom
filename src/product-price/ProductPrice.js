import React from 'react'

const ProductPrice = ({ children }) => (
  <p className="product-price">
    Price: 
    {children}
  </p>
)

export default ProductPrice