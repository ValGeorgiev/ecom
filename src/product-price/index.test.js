import React from 'react'
import ProductPrice from './index'
import renderer from 'react-test-renderer'
import { AuthContext } from '../ContextWrapper'

describe('ProductPrice component', () => {
  test('should render null if price is undefined', () => {
    const component = renderer.create(
      <ProductPrice />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should render correctly if price is 5$', () => {
    const component = renderer.create(
      <ProductPrice price={5} />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should render discount message if user is authenticated', () => {
    const component = renderer.create(
      <AuthContext.Provider value={{auth: true}}>
        <ProductPrice price={5} />
      </AuthContext.Provider>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})