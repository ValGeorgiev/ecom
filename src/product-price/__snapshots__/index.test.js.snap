// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrice component should render correctly if price is 5$ 1`] = `
<div
  className="product-price"
>
  Price:
  <span>
    5
    $
  </span>
</div>
`;

exports[`ProductPrice component should render discount message if user is authenticated 1`] = `
<div
  className="product-price"
>
  Price:
  <span>
    5
    $
  </span>
  <p>
    You have a 10% discount!
  </p>
</div>
`;

exports[`ProductPrice component should render null if price is undefined 1`] = `null`;
